### ISSUE TITLE
[Bug]: Roundstart Cyborgs do not have a brainmob associated with their MMI

### ISSUE THREAD

**Issue by Rukofamicom** [BeeStation/BeeStation-Hornet#10583]
### Describe the bug

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/9547572/b7112643-658f-46d7-ac26-91f74fc43597)

### Reproducible steps

Join the round as a cyborg (or technically as a latejoin, what's important is that it is not built from scratch)

Get disassembled in any number of ways

Runtime and stack trace triggers, player is forcibly ghosted.

**Comment by Rukofamicom** [comment]
I suspect this was introduced by #9954

@tontyGH any thoughts? I haven't given it too deep of a dive yet

**Comment by tontyGH** [comment]
Fuck

**Comment by tontyGH** [comment]
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/39193182/9dc00633-bf31-46e5-a793-bf8c8ac259e9)

**Comment by tontyGH** [comment]
The reason why it was happening was because the MMI's transfer_identity() proc called the brain_item's transfer_identity() proc expecting it to make a new brainmob for it. When Miliviu added back the mind check, it made the proc early return which caused the MMI to not have a brainmob to transfer to when it was taken out of the cyborg

**Comment by tontyGH** [comment]
tl;dr because I'm bad at coding MMI's didn't have a brainmob which obviously breaks things
